### PR TITLE
Adds weaponskill mirroring to Jailer of Prudence

### PR DESCRIPTION
**Pull request by rude-jerk**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Review by zach2good — APPROVED** [review]
Looks good to me, thanks!

**Review by claywar — APPROVED** [review]

File: src/map/ai/controllers/mob_controller.cpp [hunk 1]
```diff
@@ -955,6 +955,7 @@ bool CMobController::MobSkill(uint16 targid, uint16 wsid)
     if (POwner)
     {
         FaceTarget(targid);
+        PMob->PAI->EventHandler.triggerListener("WEAPONSKILL_BEFORE_USE", PMob, wsid);
```
Comment (TeoTwawki) [inline src/map/ai/controllers/mob_controller.cpp:958]: wondering if we should leave it weaponskill or say mobskill instead to further ifferentiate between player ws.

Comment (zach2good) [reply src/map/ai/controllers/mob_controller.cpp:958]: Do we use "MOBSKILL" in the other listener handlers? If so, that's my vote

Comment (TeoTwawki) [reply src/map/ai/controllers/mob_controller.cpp:958]: alas, t'would appear we do NOT so changing just this one would result in inconsistency.

**Pull request merged**